core/store/src/archive/cold_storage.rs

```diff
@@ -57,8 +58,7 @@ struct BatchTransaction {
 }
 
 /// Updates provided cold database from provided hot store with information about block at `height`.
-/// Returns if the block was copied (false only if height is not present in `hot_store`).
-/// Block as `height` has to be final.
+/// Block at `height` has to be final and present in `hot_store`.
```

Comment:
If it is not too hard can you assert that this is the case?

Reply:
If it is not present, the function will return error.
Regarding finality, it was asserted in `cold_store_copy()`:
```
if next_height > hot_final_head_height {
            return Err(ColdStoreError::SkippedBlocksBetweenColdHeadAndNextHeightError {
                cold_head_height,
                next_height,
                hot_final_head_height,
            });
        }
```